crates/bevy_render/src/render_resource/pipeline.rs

```diff
@@ -93,7 +93,9 @@ pub struct RenderPipelineDescriptor {
     /// Debug label of the pipeline. This will show up in graphics debuggers for easy identification.
     pub label: Option<Cow<'static, str>>,
     /// The layout of bind groups for this pipeline.
-    pub layout: Option<Vec<BindGroupLayout>>,
+    pub layout: Vec<BindGroupLayout>,
+    /// The push constant ranges for this pipeline.
+    pub push_constant_ranges: Vec<PushConstantRange>,
```

Comment:
Since there's no Default impl, I would add a comment saying that the expected default value is an empty vec.